[SheYuting]iP

SheYuting opened the pull request:
# Xuxin Chatbot Project 🌟

### Welcome to Chatbot Xuxin, 
an **interactive chatbot application** designed to manage tasks efficiently. This project supports various task-related commands, including adding, marking, deleting, and finding tasks. Below are instructions and details of the implementation.
 
Here is the launcher class:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
``` 
### Features of the Project 🛠️
Task Management: Add, delete, and modify tasks in real-time.
File Storage: Save and load tasks from a file on disk.
Command Recognition: Recognizes multiple task types (e.g., todo, deadline, event).
### How to Run the Project 🚀
[Fork](https://github.com/SheYuting/ip) the latest release in GitHub

> Open the cloned copy and run the file build.gradle:
` ./gradlew run`

### Here’s a list of commands Xuxin recognizes:

1. list: Display all tasks.
2. todo, deadline, event: Add different types of tasks.
3. delete: delete a task.
4. mark, unmark: mark or unmark the task as done.
5. find: find a task.
6. bye: exit the chatbot.

### Development Tasks ✅
 

- [x] Implement task storage functionalities.
- [x]  Add command parsing logic.
- [x]  Include Unit Tests and System Tests for error handling.
- [x]  Implement additional user commands.

Note: The project currently supports simple text-based task commands. Future enhancements may include GUI integration and additional features.

Feel free to explore the code, submit PRs, and contribute to the project! 😊

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SheYuting committed:
added greet and exit

SheYuting committed:
Added function echo

SheYuting committed:
added ability to store tasks

SheYuting committed:
Added Class Task and added functions mark and unmark task

SheYuting committed:
added class Task, Deadline, Event, Todo, updated class Duke

SheYuting committed:
added exception handling codes

SheYuting committed:
added delete function with error handling and DukeException class

SheYuting committed:
Added text-ui-test

SheYuting committed:
Added method to save the list

SheYuting committed:
Added DateTimeFormatter

SheYuting committed:
Rebuilt into a more OOP style

SheYuting committed:
Regrouped the classes into package Duke

SheYuting committed:
Merge branch 'add-gradle-support'

SheYuting committed:
Add JUnit Tests

SheYuting committed:
Added JUnit Tests using Gradle

SheYuting committed:
Added JavaDoc descriprion

SheYuting committed:
Tweak the code to comply with checkstyle

SheYuting committed:
Updated JavaDoc description

SheYuting committed:
Removed redundant method

SheYuting committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/main/Ui.java
#	src/main/java/duke/task/Deadline.java

SheYuting committed:
Added find function

SheYuting committed:
Merge branch 'branch-level-9'

SheYuting committed:
Add CheckStyle in build.gradle

SheYuting committed:
Apply GUI using FXML

SheYuting committed:
Revert "Apply GUI using FXML"

This reverts commit a3a3008bb27a6191af525c51ba92dc47fbf12852.

SheYuting committed:
Add JavaDoc comments

SheYuting committed:
Merge branch 'branch-level-10'

# Conflicts:
#	src/main/java/duke/main/MainWindow.java

SheYuting committed:
Fixed some bug

nhocmt227 commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,50 @@
+package duke.main;
```
I think you did a good job in following the Java naming convention.

nhocmt227 commented on src/main/java/duke/command/ListTaskCommand.java:
```diff
@@ -0,0 +1,18 @@
+import duke.main.Ui;
+import duke.exception.DukeException;
+
+public class ListTaskCommand extends Command {
```
Remember to include the JavaDoc before any classes or methods in your code for better code clarity.

nhocmt227 commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,33 @@
+    public static Command parse(String fullCommand) throws DukeException {
+        String commandWord = fullCommand.split(" ")[0];
+
+        switch (commandWord) {
```
Perhaps you can put `case` and `switch` to have the same indentation to better align with the Java coding convention.

DengTianle replied:
Agree. Perhaps you can delete the indentation for `case` as per the course coding standard? If you are using IntelliJ, you can follow [this](https://se-education.org/guides/tutorials/intellijCodeStyle.html#tweak-switch-case-style) guide to change the default style.

nhocmt227 reviewed (commented):
PR review 2

DengTianle commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,42 @@
+    private String date;
+
+    public AddDeadlineCommand(String command) throws DukeException {
+        if (!command.contains(" /by ")) throw new DukeException("A deadline must have '/by '.");
```
Perhaps you could consider always putting the conditional on a separate line and wrapping it, as per the course coding standard?

DengTianle reviewed (commented):
PR review 1 (coding standard)
I like how clean and understandable your code is, good job!

SheYuting committed:
Modify some coding standards

In the Parser class, the switch case method was used with a wrong
indentation level. JavaDoc comments are also missing in some public
classes.

Modify all those with correct standards which improves the code quality.

SheYuting committed:
Currently, the classes do not have assertions to verify assumptions
about method parameters and object states.

This can lead to unexpected errors if invalid inputs are passed.

Add assertions to verify that the input parameters are valid.

SheYuting committed:
Merge pull request #1 from SheYuting/branch-A-Assertions

Currently, the classes do not have assertions to verify assumptions

SheYuting committed:
Merge branch 'master' into branch-A-CodeQuality

SheYuting committed:
Merge pull request #2 from SheYuting/branch-A-CodeQuality

Modify some coding standards

SheYuting committed:
Added feature statistics

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Merge branch 'master' into patch-1

SheYuting committed:
Merge pull request #3 from SheYuting/patch-1

Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Fix coding error

SheYuting committed:
Revert "Fix coding error"

This reverts commit c7a0747e005934ce554924862b116538e740b6df.

SheYuting committed:
Fix coding standard violations

SheYuting committed:
Imporve GUI design

SheYuting committed:
Add personality and more command

SheYuting committed:
Change bot name and refine formatting

SheYuting committed:
Change bot name

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Add files via upload

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update tests

SheYuting committed:
Add files via upload

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Update README.md

SheYuting committed:
Add more JavaDoc

SheYuting committed:
Update Ui picture

SheYuting committed:
Modify date time format
